documentation/sphinx/source/developer-guide.rst

```diff
@@ -463,7 +463,6 @@ In some situations, you may want to explicitly control the number of key-value p
 
   LIMIT = 100 # adjust according to data characteristics
 
-  @fdb.transactional
```

Comment:
This same code block exists in segmented-range-reads.rst, so it will need to be changed there too. I'm also a little uncertain about what this function should actually do and whether it makes sense to just take off the `transactional` decorator. In general, our code here should look like what we want people to write, so we should try to decide what the use-case for this code really is and how we would write code to meet that need.

By the way, part of my confusion about what we're trying to do here is that we also have a Java implementation of this function (in segmented-range-reads-java.rst) that behaves a little bit differently (and is also a little broken, I think).

Reply:
That code is fine, as long as it's called with a transaction and not a database.  So, removing the @fdb.transactional so that it's just always passed a transaction makes it do the right thing.

I could rewrite it to be closure-based with a return if you'd prefer.